controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -310,8 +310,54 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases() (reconcile.Resul
 		// Update the StorageClassClaim status.
 		r.storageClassClaim.Status.Phase = v1alpha1.StorageClassClaimReady
 
+	} else {
+
+		// Reconciliation needed if the StorageClassClaim is present.
+		if err = r.get(r.storageClassClaim); !errors.IsNotFound(err) {
+
+			// Deletion phase.
+			// Update the StorageClassClaim status.
+			r.storageClassClaim.Status.Phase = v1alpha1.StorageClassClaimDeleting
+
+			// Delete StorageClass.
+			existingStorageClass := &storagev1.StorageClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: r.storageClassClaim.Name,
+				},
+			}
+			if err = r.get(existingStorageClass); err == nil {
+
+				// Remove finalizer from StorageClass.
+				existingStorageClass.SetFinalizers(remove(existingStorageClass.Finalizers, StorageClassFinalizer))
+				if err := r.update(existingStorageClass); err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to remove finalizer from StorageClass: %s", err)
+				}
+				if err = r.delete(existingStorageClass); err != nil {
```

Comment:
I think there is an issue with the order of operations. 
I think we should not delete the storageclass until we got an ack on the revoke operation.